models/ml-features.yaml

```diff
@@ -55,250 +55,233 @@ models:
         - entity_var:
             from: inputs/stripeCharges
             name: customer_currency_ts
-            select: max(created)
-            where: currency is not null and currency != ''
+            select: max({{stripeCharges.Var('created')}})
+            where: "{{stripeCharges.Var('currency')}} is not null and {{stripeCharges.Var('currency')}} != ''"
         - entity_var:
             from: inputs/stripeCharges
             name: customer_currency
-            select: max(currency)
-            where: created = customer_currency_ts
+            select: max({{stripeCharges.Var('currency')}})
+            where: "{{stripeCharges.Var('created')}} = {{user.Var('customer_currency_ts')}}"
         - entity_var:
             from: inputs/stripeCharges
             name: payment_method_details_card_network_ts
-            select: max(created)
+            select: max({{stripeCharges.Var('created')}})
             where: currency is not null and currency != ''
         - entity_var:
             from: inputs/stripeCharges
             name: payment_method_details_card_network
             select: max(payment_method_details_card_network)
-            where: created = payment_method_details_card_network_ts
+            where: "{{stripeCharges.Var('created')}} = {{user.Var('payment_method_details_card_network_ts')}}"
         - entity_var:
             from: inputs/stripeCustomers
             name: is_delinquent
-            select: max(delinquent)
+            select: max({{stripeCustomers.Var('created')}})
         - entity_var:
             from: inputs/stripeCharges
             name: has_credit_card
-            select: max(case when (payment_method_details_card_funding = 'credit' and status = 'succeeded') then 1 else 0 end)
+            select: max(case when ({{stripeCharges.Var('payment_method_details_card_funding')}} = 'credit' and {{stripeCharges.Var('status')}} = 'succeeded') then 1 else 0 end)
         - entity_var:
             from: inputs/stripeCharges
             name: first_sale_time
-            select: min(created)
+            select: min({{stripeCharges.Var('created')}})
             where: status = 'succeeded'
         - entity_var:
             name: days_since_first_sale
-            select: "{{ macro_datediff('first_sale_time') }}"
-            dependencies:
-              - first_sale_time
+            select: "{{ macro_datediff('{{user.Var(\"first_sale_time\")}}') }}"
         - entity_var:
             from: inputs/stripeCharges
             name: last_purchase_time
-            select: max(created)
+            select: max({{stripeCharges.Var('created')}})
             where: status = 'succeeded'
         - entity_var:
             name: days_since_last_purchase
-            select: "{{ macro_datediff('last_purchase_time') }}"
+            select: "{{ macro_datediff('{{user.Var(\"last_purchase_time\")}}') }}"
             dependencies:
```

Comment:
@yajuaryarudderstack we don't need dependencies if we have the user.Var right?

Reply:
@dpatchigolla Yes, if we are using <entityName>.Var then no need to use dependencies, it's redundant.